src/gestionstock_gt/Accueil.java

```diff
@@ -824,6 +1347,99 @@ private void btnSearchMouseClicked(java.awt.event.MouseEvent evt) {//GEN-FIRST:e
                 }
     }//GEN-LAST:event_btnSearchMouseClicked
 
+    private void btn_add_consActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_btn_add_consActionPerformed
+        // TODO add your handling code here:
+           int id=cons.getKey(select_prod_cons.getSelectedItem().toString());
+        //System.out.println("(key "+id+", value "+select_produit_cons.getSelectedItem()+")");
+        
+        if(id>0 && zone_qte_cons.getText()!="")
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.